cmd/broker/fanout/main.go

```diff
@@ -99,7 +104,7 @@ func main() {
 	if err != nil {
 		logger.Fatal("Failed to create fanout sync pool", zap.Error(err))
 	}
-	if _, err := handler.StartSyncPool(ctx, syncPool, syncSignal, env.MaxStaleDuration, handler.DefaultProbeCheckPort); err != nil {
+	if _, err := handler.StartSyncPool(ctx, syncPool, syncSignal, env.MaxStaleDuration, handler.DefaultProbeCheckPort, env.AuthType, authcheckclient.NewAuthCheckClient()); err != nil {
```

Comment:
It doesn't impact the functionality for sure, but I'm curious why not wrapping all authcheck stuff into a single interface? E.g.

```
type AuthChecker interface {
	Check() error
}
```

The real impl could contain the `AuthType` and the fake impl just needs to implement the `Check()` func.

In this way, you don't have to check the `StartSyncPool` to pass along the auth type.

Reply:
Are you referring to inject those parameters when InitializeSyncPool instead of passing the parameter into StartSyncPool, or referring to add an interface into startSyncPool instead of those two parameters? I still need to pass the authType into the pool in order to make the auth check work together with existing probe checkers. And even I use an interface, I still need the mock client for my other UTs (to add UT for authentication check in authcheck.go). So I just re-use the mock client for pool related UTs (If I add the interface, I'll need additional fake interface for those UTs in addition to the mock client). I can change to use an interface if you think that is more proper.

Reply:
done. changed to use interface